troposphere/static/js/actions/instance/resume.js

```diff
@@ -1,58 +1,90 @@
+ */
+function handleOne(instance) {
+    // if we've been passed a model via "selected resources" that
+    // is *not* an Instance, then just skip it - _move on_
+    if (!(instance instanceof Instance)) return;
+
+    let instanceState = new InstanceState({
+            status_raw: "suspended - resuming",
+            status: "active",
+            activity: "resuming"
+        }),
+        originalState = instance.get("state"),
+        actionRequest = new InstanceActionRequest({
+            instance: instance
+        });
+
+    instance.set({
+        state: instanceState
+    });
+
+    Utils.dispatch(InstanceConstants.UPDATE_INSTANCE, {
+        instance: instance
+    });
+
+    actionRequest.save(null, {
+        attrs: {
+            action: "resume"
+        }
+    }).done(function() {
+        // Consider "success" notification ...
         instance.set({
             state: instanceState
         });
+    }).fail(function(response) {
+        instance.set({
+            state: originalState
+        });
+        Utils.displayError({
+            title: "Your instance could not be resumed",
+            response: response
+        });
+    }).always(function() {
         Utils.dispatch(InstanceConstants.UPDATE_INSTANCE, {
             instance: instance
         });
-
-        actionRequest.save(null, {
-            attrs: {
-                action: "resume"
-            }
-        }).done(function() {
-            instance.set({
-                state: instanceState
-            });
-        }).fail(function(response) {
-            instance.set({
-                state: originalState
-            });
-            Utils.displayError({
-                title: "Your instance could not be resumed",
-                response: response
-            });
-        }).always(function() {
-            Utils.dispatch(InstanceConstants.UPDATE_INSTANCE, {
-                instance: instance
-            });
-            Utils.dispatch(InstanceConstants.POLL_INSTANCE_WITH_DELAY, {
-                instance: instance,
-                delay: 25*1000,
-            });
+        Utils.dispatch(InstanceConstants.POLL_INSTANCE_WITH_DELAY, {
+            instance: instance,
+            delay: 25*1000,
         });
+    });
+}
+
+
+export default {
+
+    resume: function(params) {
+        if (!params.resources)
+            throw new Error("Missing resources");
+        if (params.instance)
+            throw new Error("Deprecated parameter passed");
+
+        // We expect `resources` to be a backbone collection
+        let resources = params.resources;
+
+        // in case "resources" is not a collection, but is
+        // an Instance passed as `resources`, make a collection
+        // and add the "models/Instance" to it
+        if (resources instanceof Instance) {
+            let instance = resources;
+            resources = new Backbone.Collection(instance);
+        }
+
+        resources.map(handleOne);
```

Comment:
Well, my concern is that it might stick-out or appear awkward without context. 

I think this at least gets us to a place where _actions_ can be re-worked in the future. A move to _Redux_ would like mean gutting this code. 

I guess you might say the problem with repetitious code is that it stands on its' own within a single module - but it a nightmare to maintain.